### ISSUE TITLE
multi-band Melgan training

### ISSUE THREAD

**Issue by unparalleled-ysj** [TensorSpeech/TensorflowTTS#115]
Hi, when I use my own data to train mb-melgan, there is always a strong current sound at the end of the generated speech, is it because there are not enough iterations?
![64](https://user-images.githubusercontent.com/39238619/87525703-71959080-c6bc-11ea-93d1-4de7b502087b.png)

**Comment by ZhaoZeqing** [comment]
I also trained multi-band melgan using my own data, sr is 22.05k, all conf files are same to LJSpeech, in the predictions dir, 5000steps is normal,  but starting from 10000steps, the generated audio is weried, like this:

![64](https://user-images.githubusercontent.com/8464475/87529719-fafb9180-c6c1-11ea-816b-5f7be5f77154.png)

![64](https://user-images.githubusercontent.com/8464475/87529738-ffc04580-c6c1-11ea-90c4-47ac86e7dbdc.png)

![63](https://user-images.githubusercontent.com/8464475/87529775-1070bb80-c6c2-11ea-8d4b-d77b90bc6446.png)

**Comment by dathudeptrai** [comment]
@ZhaoZeqing @unparalleled-ysj all the things report here already has solution at #113

**Comment by ZhaoZeqing** [comment]
@dathudeptrai I will try the solution! Thanks :)

**Comment by dathudeptrai** [comment]
@unparalleled-ysj let me explained why ur problem occur:

1. strong current sound at the end is the padding sound. Let say our mel normalize based-on mean/std so the mel range is around -4 -> 4. If you norm the mel from 0->4 then 0 is the value of **silience voices**. But in the training progress, the padding mel is 0 then it's not a **silience voice** so it cause ur problem.
2. it's not important when you inference on the practice since you know the mel length then you also know the real length of the audio.
3. But if you want to solve this problem, you can refer my solution at #113 :))), i will commit the solution soon with many other improvements :d

**Comment by unparalleled-ysj** [comment]
> @unparalleled-ysj let me explained why ur problem occur:
> 
> 1. strong current sound at the end is the padding sound. Let say our mel normalize based-on mean/std so the mel range is around -4 -> 4. If you norm the mel from 0->4 then 0 is the value of **silience voices**. But in the training progress, the padding mel is 0 then it's not a **silience voice** so it cause ur problem.
> 2. it's not important when you inference on the practice since you know the mel length then you also know the real length of the audio.
> 3. But if you want to solve this problem, you can refer my solution at #113 :))), i will commit the solution soon with many other improvements :d

I got it , thank you for your detailed reply !

**Comment by ZhaoZeqing** [comment]
@dathduptrai  I tried the solution at #113 , the loss explosion still occurred , but when I just modified https://github.com/TensorSpeech/TensorflowTTS/blob/master/examples/multiband_melgan/conf/multiband_melgan.v1.yaml#L69 to batch_max_steps: 16384,  it worked. Now it trained at 55000 steps :D

**Comment by dathudeptrai** [comment]
@ZhaoZeqing you can also try reduce learning-rate by replace the first value of https://github.com/TensorSpeech/TensorflowTTS/blob/master/examples/multiband_melgan/conf/multiband_melgan.v1.yaml#L82 by 0.0005 :)). I think **STFT loss** has some data-related problems which can be solved by **reduce learning-rate**, **Clip loss** or **Increase batch_max_steps** :)).
